Added DaoJdbcImpl for bucket, user and order

TatianaPirch opened the pull request:
I will not be in time today to transfer work with roles to RoleDao and remake methods so that they use Optional . Upload as promised until 19:00

TatianaPirch committed:
Added DaoJdbcImpl for bucket, user and order

boroda4436 commented on src/main/java/mate/academy/internetshop/dao/impl/UserDaoImpl.java:
```diff
@@ -29,7 +28,7 @@ public User get(Long id) {
     public User update(User user) {
         User updatedUser = get(user.getId());
-        updatedUser.setOrders(user.getOrders());
+        //updatedUser.setOrders(user.getOrders());
```
@TatianaPirch please remove comments

boroda4436 commented on src/main/java/mate/academy/internetshop/dao/impl/BucketDaoImpl.java:
```diff
@@ -34,16 +34,36 @@ public Bucket getBucket(Long userId) {
 
+    @Override
+    public Bucket addItem(Long bucketId, Long itemId) {
+        return null;
```
@TatianaPirch звичайно було б круто імплементувати ці методи також

boroda4436 commented on src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java:
```diff
@@ -0,0 +1,151 @@
+        super(connection);
+    }
+
+    public List<Item> addItemsInDb(Order order) {
```
```suggestion
    private List<Item> addItemsInDb(Order order) {
```
@TatianaPirch can this method be private?

boroda4436 commented on src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java:
```diff
@@ -0,0 +1,151 @@
+        return items;
+    }
+
+    public List<Item> getItemsFromDb(Long orderId) {
```
@TatianaPirch can this method be private?
```suggestion
    private List<Item> getItemsFromDb(Long orderId) {
```

boroda4436 commented on src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java:
```diff
@@ -0,0 +1,151 @@
+        return items;
+    }
+
+    public void deleteItemFromDB(Long orderId) {
```
```suggestion
    private void deleteItemFromDB(Long orderId) {
```
@TatianaPirch can this method be private?

boroda4436 commented on src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java:
```diff
@@ -0,0 +1,211 @@
+        super(connection);
+    }
+
+    public Set<Role> addRoleToDB(User user) {
```
```suggestion
    private Set<Role> addRoleToDB(User user) {
```
@TatianaPirch can this method be private?

boroda4436 commented on src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java:
```diff
@@ -0,0 +1,211 @@
+        return roles;
+    }
+
+    public void deleteRoleToDB(Long userId) {
```
```suggestion
    private void deleteRoleToDB(Long userId) {
```

boroda4436 commented on src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java:
```diff
@@ -0,0 +1,211 @@
+        }
+    }
+
+    public Set<Role> getRolesDb(User user) {
```
@TatianaPirch 
```suggestion
    private Set<Role> getRolesDb(User user) {
```

boroda4436 commented on src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java:
```diff
@@ -0,0 +1,211 @@
+                String password = resultSet.getString("password");
+                User user = new User(name, surname, login, password);
+                user.setId(id);
+                user.setToken(token);
```
@TatianaPirch is this code repeatable in several places in this class? if yes - is it possible move it to separate method?

TatianaPirch committed:
1. OrderDaoJdbcImpl : made helper methods private.
2. Created RoleDao and RoleDaoJdbcImpl.
3. Transferred methods for roles in RoleDao.
4. BucketDaoImpl : implemented all methods.
5. UserDaoJdbcImpl: changed  implementation for delete method,
created separated method userBuild
